### PR TITLE
[Perf] Use numpy zero-copy path for embedding float response serialization

### PR DESCRIPTION
**Pull request by lokashrinav**
## Summary

When `encoding_format=float` and `ORJSONResponse` is active, the `/v1/embeddings` response builder currently calls `.tolist()` on each embedding tensor, converting every float individually through Python. This PR replaces that with a zero-copy `.numpy()` view that ORJSON serializes natively.

**Scope:** Only the float-format OpenAI embedding response path is changed. Base64, bytes, Cohere, non-ORJSON fallback, model execution, scheduling, and KV cache are untouched.

**Fallback:** If `.numpy()` fails (bfloat16, float8, CUDA tensors), it falls back to the existing `.tolist()` path.

## Benchmark results (Modal, T4 GPU, vLLM 0.20.1, intfloat/multilingual-e5-small)

Response construction microbenchmark (batch=128):

| Embedding dim | `.tolist()` (ms) | `.numpy()` (ms) | Speedup |
|---:|---:|---:|---:|
| 384 | 1.43 | 0.21 | **6.8x** |
| 768 | 3.97 | 0.22 | **18.2x** |
| 1536 | 8.67 | 0.22 | **39.9x** |
| 3072 | 14.63 | 0.21 | **68.8x** |

The new path is O(1) (zero-copy view) vs O(batch × dim) for `.tolist()`.

## E2E validation

Tested on Modal (T4, vLLM 0.20.1 with patch applied):
- Single and batch `/v1/embeddings` requests with `encoding_format=float` ✓
- `encoding_format=base64` (unchanged path) ✓
- `EmbeddingResponse.model_validate()` on fast-path output ✓
- Float vs base64 cross-check (values match within tolerance) ✓
- Latency (batch-16): p50=26.9ms, p95=29.8ms

## Test plan

- [x] Unit tests for `encode_pooling_output_float_or_ndarray` (float32 → ndarray, fallback on TypeError)
- [x] E2E server test with real model (intfloat/multilingual-e5-small, fp16)
- [ ] Upstream CI (`pytest tests/entrypoints/pooling/test_utils.py`)
- [ ] Existing embedding tests pass (`pytest tests/entrypoints/pooling/embed/test_online.py`)

### CODE REVIEW

**Comment by lokashrinav** [comment]
Addressed in a647038 — the fast path now uses `EmbeddingResponseData.model_dump()` and `EmbeddingResponse.model_dump()` to build the response structure, then injects the numpy arrays into the resulting dicts. This keeps the response schema in sync with the Pydantic models while preserving the zero-copy serialization benefit.

Re-verified on Modal (T4 GPU, vLLM 0.20.1 + patch, intfloat/multilingual-e5-small):
- All unit tests pass
- All E2E tests pass (single float, batch float, base64, schema validation, float↔base64 cross-check)
- Performance unchanged (5–70x speedup in response construction)

File: tests/entrypoints/pooling/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import json
+import warnings
+from types import SimpleNamespace
+
+import numpy as np
+import pytest
+import torch
+from fastapi.responses import ORJSONResponse
+
+from vllm.entrypoints.pooling.utils import encode_pooling_output_float_or_ndarray
```
Comment (noooop) [inline tests/entrypoints/pooling/test_utils.py:14]: Thanks for the optimization.

I think we don't have ORJSON installed, so this test will fail.

**Review by noooop — APPROVED** [review]
Thanks for your improvement!
